src/Collection.php

```diff
@@ -508,10 +507,7 @@ static function (Generator $generator): Generator {
      */
     public static function fromIterable(iterable $iterable): self
     {
-        return new self(
-            static fn (iterable $iterable): Iterator => new IterableIterator($iterable),
-            [$iterable]
-        );
+        return new self(static fn (): Iterator => new IterableIterator($iterable));
```

Comment:
does it matter if we pass the parameter to the closure or we just let it capture it from the environment? At least with the testing we have it doesn't seem to make any difference, but it fixes a PHPStan error